ide/refactoring.api/src/org/netbeans/modules/refactoring/spi/impl/RefactoringPanel.java

```diff
@@ -381,16 +385,26 @@ private JToolBar getToolBar() {
         toolbar.add(expandButton);
         toolbar.add(logicalViewButton);
         toolbar.add(physicalViewButton);
-        if (ui instanceof RefactoringCustomUI) {
+        toolbar.add(previewButton);
+        if (refactoringUI instanceof RefactoringCustomUI) {
             toolbar.add(customViewButton);
         }
         return toolbar;
     }
-    
+
+    private Preferences getPreferences() {
+        return NbPreferences.forModule(RefactoringPanel.class);
+    }
+
+    private String preferencesKeyForUI(String uiPreference) {
+        Class<?> targetClass = refactoringUI.getClass();
+        return targetClass.getName() + "." + uiPreference;
+    }
```

Comment:
this is just a nitpick but there are about 20 `RefactoringUI` subclasses - I am wondering if this kind of granularity is needed. It might look like a bug to the user that the rename refactoring preview setting isn't the same as the add-override refactoring setting.

It might potentially be better to use something like
```java
getClass().getName() + "_" + (isQuery ? "query" : "refactoring") + "." + uiPreference;
```
 as key. The windows should be fairly similar to each other, the main difference is diff vs regular preview.

Reply:
Good idea. Adjusted.